File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/registruIntrariIesiri/RegistruIesiriModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.registruIntrariIesiri;

import java.util.Date;
import java.util.List;

import ro.cloudSoft.cloudDoc.domain.AtasamentModel;
import ro.cloudSoft.cloudDoc.presentation.client.shared.model.project.SubactivityModel;

public class RegistruIesiriModel {

	private Long id;
	private String numarInregistrare;
	private Date dataInregistrare;
	private Long tipDocumentId;
	private Integer numarPagini;
	private Integer numarAnexe;
	private Long intocmitDeUserId;
	private boolean trimisPeMail;
	private String continut;
	private boolean asteptamRaspuns;
	private Date termenRaspuns;
	private List<RegistruIesiriDestinatarModel> destinatari;
	private List<Long> proiectIds;
	private boolean anulat;
	private String motivAnulare;
	private boolean inchis;
	private boolean trebuieAnulat;
	private List<AtasamentModel> atasamente;
	private SubactivityModel subactivity;

	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public String getNumarInregistrare() {
		return numarInregistrare;
	}
	public void setNumarInregistrare(String numarInregistrare) {
		this.numarInregistrare = numarInregistrare;
	}
	public Date getDataInregistrare() {
		return dataInregistrare;
	}
	public void setDataInregistrare(Date dataInregistrare) {
		this.dataInregistrare = dataInregistrare;
	}
	public Long getTipDocumentId() {
		return tipDocumentId;
	}
	public void setTipDocumentId(Long tipDocumentId) {
		this.tipDocumentId = tipDocumentId;
	}
	public Integer getNumarPagini() {
		return numarPagini;
	}
	public void setNumarPagini(Integer numarPagini) {
		this.numarPagini = numarPagini;
	}
	public Integer getNumarAnexe() {
		return numarAnexe;
	}
	public void setNumarAnexe(Integer numarAnexe) {
		this.numarAnexe = numarAnexe;
	}
	public Long getIntocmitDeUserId() {
		return intocmitDeUserId;
	}
	public void setIntocmitDeUserId(Long intocmitDeUserId) {
		this.intocmitDeUserId = intocmitDeUserId;
	}
	public boolean isTrimisPeMail() {
		return trimisPeMail;
	}
	public void setTrimisPeMail(boolean trimisPeMail) {
		this.trimisPeMail = trimisPeMail;
	}
	public String getContinut() {
		return continut;
	}
	public void setContinut(String continut) {
		this.continut = continut;
	}
	public boolean isAsteptamRaspuns() {
		return asteptamRaspuns;
	}
	public void setAsteptamRaspuns(boolean asteptamRaspuns) {
		this.asteptamRaspuns = asteptamRaspuns;
	}
	public Date getTermenRaspuns() {
		return termenRaspuns;
	}
	public void setTermenRaspuns(Date termenRaspuns) {
		this.termenRaspuns = termenRaspuns;
	}
	public List<RegistruIesiriDestinatarModel> getDestinatari() {
		return destinatari;
	}
	public void setDestinatari(List<RegistruIesiriDestinatarModel> destinatari) {
		this.destinatari = destinatari;
	}
	public List<Long> getProiectIds() {
		return proiectIds;
	}
	public void setProiectIds(List<Long> proiectIds) {
		this.proiectIds = proiectIds;
	}
	public boolean isAnulat() {
		return anulat;
	}
	public void setAnulat(boolean anulat) {
		this.anulat = anulat;
	}
	public String getMotivAnulare() {
		return motivAnulare;
	}
	public void setMotivAnulare(String motivAnulare) {
		this.motivAnulare = motivAnulare;
	}
	public boolean isInchis() {
		return inchis;
	}
	public void setInchis(boolean inchis) {
		this.inchis = inchis;
	}
	public List<AtasamentModel> getAtasamente() {
		return atasamente;
	}
	public void setAtasamente(List<AtasamentModel> atasamente) {
		this.atasamente = atasamente;
	}
	public boolean isTrebuieAnulat() {
		return trebuieAnulat;
	}
	public void setTrebuieAnulat(boolean trebuieAnulat) {
		this.trebuieAnulat = trebuieAnulat;
	}
	public SubactivityModel getSubactivity() {
		return subactivity;
	}
	public void setSubactivity(SubactivityModel subactivity) {
		this.subactivity = subactivity;
	}
	
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/registru-intrari-iesiri/registru-iesiri.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { RegistruIesiriDestinatariModel } from "./registru-iesiri-destinatari.model";
import { JsonDateConverter } from "../../json-mapper";
import { RegistruIesiriAtasamentModel } from "./registru-iesiri-atasamente.model";
import { ProjectSubactivityModel } from "../project/project-subactivity.model";

@JsonObject
export class RegistruIesiriModel {

	@JsonProperty("id", Number)
	public id: number = null;

	@JsonProperty("numarInregistrare", String)
	public numarInregistrare: string = null;
	
	@JsonProperty("dataInregistrare", JsonDateConverter)
	public dataInregistrare: Date = null;
	
	@JsonProperty("tipDocumentId", Number)
	public tipDocumentId: number = null;

	@JsonProperty("numarPagini", Number)
	public numarPagini: number = null;

	@JsonProperty("numarAnexe", Number)
	public numarAnexe: number = null;

	@JsonProperty("intocmitDeUserId", Number)
	public intocmitDeUserId: number = null;

	@JsonProperty("trimisPeMail", Boolean)
	public trimisPeMail: boolean = false;

	@JsonProperty("continut", String)
	public continut: string = null;

	@JsonProperty("asteptamRaspuns", Boolean)
	public asteptamRaspuns: boolean = false;

	@JsonProperty("termenRaspuns", JsonDateConverter)
	public termenRaspuns: Date = null;

	@JsonProperty("destinatari", [RegistruIesiriDestinatariModel])
	public destinatari: RegistruIesiriDestinatariModel[] = [];

	@JsonProperty("proiectIds", [Number])
	public proiectIds: number[] = [];

	@JsonProperty("anulat", Boolean)
	public anulat: boolean = false;

	@JsonProperty("inchis", Boolean)
	public inchis: boolean = false;

	@JsonProperty("motivAnulare", String)
	public motivAnulare: string = null;
	
	@JsonProperty("atasamente", [RegistruIesiriAtasamentModel])
	public atasamente:  RegistruIesiriAtasamentModel[] = [];
	
	@JsonProperty("subactivity", ProjectSubactivityModel)
	public subactivity: ProjectSubactivityModel = null;
}